pkgs/applications/version-management/yadm/default.nix

```diff
@@ -1,17 +1,17 @@
 { lib, stdenv, fetchFromGitHub, git, gnupg }:
 
-let version = "2.5.0"; in
+let version = "3.1.0"; in
 stdenv.mkDerivation {
   pname = "yadm";
   inherit version;
 
   buildInputs = [ git gnupg ];
 
   src = fetchFromGitHub {
-    owner  = "TheLocehiliosan";
-    repo   = "yadm";
-    rev    = version;
-    sha256 = "128qlx8mp7h5ifapqqgsj3fwghn3q6x6ya0y33h5r7gnassd3njr";
+    owner = "TheLocehiliosan";
+    repo = "yadm";
+    rev = "a4d39c75045bfca9277284ff0513eb022237a88e";
```

Comment:
```suggestion
    rev = version;
```

Reply:
Ah, thanks I made this change because of user error on my part. The reason seems like a separate issue but I'll write it here in case my understanding is wrong.

I thought `rev` specified the version to download and `sha256` was a hash to check that the download was correct. For some reason, on my machine, if `rev = 3.1.0` and I keep the sha the same (for the 2.5 version) it will download the 2.5 source, and accept that source code as correct. If I switch the sha to the 3.1 sha, and keep everything else the same, it will download the 3.1 code and accept that. I thought I was going crazy, so I uninstalled the packages, ran `nix-collect-garbage` between builds and got the same behavior.

Anyway, I made this change because I assumed setting rev to a hash would fix this, but it doesn't. Going back to the version number since the behavior is the same.

Reply:
@blargg Common snag. 

The rev does specify what version to download, but the store check is based on the hash. If you already have a build from that hash in your store, it'll use that copy. I usually just change a character in the hash to `0`, but I think you can also temporarily set it to `lib.fakeHash` or `lib.fakeSha256`

Reply:
Thank you, that's good advice. I'll use these in the future :smile: